### ISSUE TITLE
Decrease neural network size to 7 intermediate layers

### ISSUE THREAD

**Issue by Chicoryn** [Chicoryn/dream-go#4]
To increase inference and training performance decrease the number of intermediate layers to 7 instead of the current 19. This should give about a 2.71x speed-up, but such a small network might not be able to learn anything useful.

**Comment by Chicoryn** [comment]
With a batch size of 1024 and learning rate starting at 3e-3 the training process is currently at the following values after about 100,000 steps:

* Policy accuracy: 47.2% / 72.5% / 80.9%
* Value accuracy: 65.5%
* Loss (total/policy/value): 2.97 / 1.93 / 0.83

These numbers are surprisingly fine and somewhat comparable to the full 19 layer network, some of this might be attributed to the large batch size, low learning rate, and increased regulation coefficient. So regardless of the final results some of these hyper-parameters might be worth carrying over to the normal network.

**Comment by Chicoryn** [comment]
After matching this newly trained network against the 19 layer network, a final score of `15 - 4` in favour of the 19 layer network was obtained. This is too large of a difference to accept the smaller network as the best.

Looking at some of the games, what seems to be happening is that the networks gets into a fight, but the 7 layer network loses once the fight gets larger than its peripheral vision. If this is true, then we cannot run any network size smaller than 19 intermediate layers as that would result is this problem appearing again only at a larger scale.

[dg-19-vs-7.zip](https://github.com/Chicoryn/dream-go/files/1591570/dg-19-vs-7.zip)
